File: OddWorksDeviceSDK/src/main/java/io/oddworks/device/metric/OddVideoErrorMetric.java
package io.oddworks.device.metric;

public class OddVideoErrorMetric extends OddMetric {
    private static final String TAG = OddVideoErrorMetric.class.getSimpleName();

    private static String videoErrorAction = ACTION_VIDEO_ERROR;
    private static boolean videoErrorEnabled = false;

    public OddVideoErrorMetric() {

    }

    public static void setAction(String action) {
        videoErrorAction = action;
    }

    public static void setEnabled(boolean enabled) {
        videoErrorEnabled = enabled;
    }

    @Override
    public String getAction() {
        return videoErrorAction;
    }

    @Override
    public boolean getEnabled() {
        return videoErrorEnabled;
    }
}

File: device-sdk/src/main/java/io/oddworks/device/metric/OddVideoErrorMetric.kt
package io.oddworks.device.metric

import org.json.JSONObject

class OddVideoErrorMetric(contentType: String, contentId: String, title: String, meta: JSONObject? = null) : OddMetric(contentType, contentId, title, meta) {
    override val action: String
        get() = OddVideoErrorMetric.action

    override val enabled: Boolean
        get() = OddVideoErrorMetric.enabled

    companion object {
        private val TAG = OddUserNewMetric::class.java.simpleName

        var action = OddMetric.ACTION_VIDEO_ERROR
        var enabled = false
    }
}